### ISSUE TITLE
feat(cli): basin-bundle loader → ScenarioBundle; fill north_atlantic / med_sea / gulf_stream / southern_ocean

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/somax#208]
## Problem / Request

Nothing loads `data/basin/<name>.zarr` into a `ScenarioBundle`. The four real-geography scenario modules are `NotImplementedError` stubs. Add one loader and make the stubs use it.

## User Story

> As a somax user, I want `scenario: {name: gulf_stream}` in a run YAML to resolve to a `ScenarioBundle` with `Geometry(kind="real_basin", mask=…, bathymetry=…, Lx, Ly)`, `Constants(f0, beta)` from the bundle, and `ForcingFields(tau_x, tau_y, heat)`, so that #78 / #79 become configuration, not code.

## Motivation

This is the missing seam identified in the audit: the bundle is written, the `ScenarioBundle` types exist (`scenarios/_types.py:102-136`), and nothing connects them.

## Proposed API

```python
# somax/_src/cli/_basin.py
def load_basin_bundle(name: str, *, data_dir: Path = Path("data/basin")) -> ScenarioBundle:
    ds = _open_bundle(data_dir / f"{name}.zarr")          # xr.open_zarr; validate basin_bundle_version + required vars
    geometry = Geometry(kind=ds.attrs["geometry_kind"], nx=ds.sizes["x"], ny=ds.sizes["y"],
                        Lx=ds.attrs.get("Lx_m"), Ly=ds.attrs.get("Ly_m"),
                        lon_bounds=tuple(ds.attrs["lon_bounds"]), lat_bounds=tuple(ds.attrs["lat_bounds"]),
                        mask=jnp.asarray(ds["mask"].values), bathymetry=jnp.asarray(ds["bathymetry"].values))
    constants = Constants(f0=float(ds.attrs["f0"]), beta=float(ds.attrs["beta"]))
    forcing = ForcingFields(tau_x=jnp.asarray(ds.tau_x.values), tau_y=..., heat=...)
    return ScenarioBundle(name=name, geometry=geometry, constants=constants, forcing=forcing, initial_condition=...)
```

Each stub's `_build(params)` becomes `load_basin_bundle(NAME)` plus the `initial_condition` / `forcing_params` handling `double_gyre` already has. If the bundle is missing, raise with the `somax-sim data fetch <name>` hint.

## References & Existing Code

- `somax/_src/cli/scenarios/{north_atlantic,med_sea,gulf_stream,southern_ocean}.py`, `_types.py`, `double_gyre.py:300-390`
- `somax/_src/cli/_data.py:271` (`_basin_zarr_path`)
- `somax/_src/cli/models_registry/spherical_swm.py:18-121` (mask / forcing padding the adapters do next)
- #78, #79 (scope), #72 decision 2 (scenarios own constants)

## Implementation Steps

- [ ] `somax/_src/cli/_basin.py` loader (xarray import inside function; `sim` group)
- [ ] Replace the four stubs; keep `global_ocean` refused via `INCOMPATIBLE_PAIRS`
- [ ] Authoring configs + DVC stages for `gulf_stream + multilayer_nonlinear_swm`, `north_atlantic + multilayer_qg`, `med_sea + nonlinear_swm`, `southern_ocean + spherical_qg`
- [ ] Tests

## Definition of Done

- [ ] `somax-sim list-scenarios` shows the four basins as runnable
- [ ] End-to-end smoke test per scenario at the synthetic bundle (`--debug`)
- [ ] `make test-cov`, lint, typecheck pass

## Testing

- [ ] Unit test: loader on the synthetic bundle → shapes, dtypes, `f0/beta` read from attrs
- [ ] Unit test: missing bundle → actionable error
- [ ] Smoke: each `(basin, model)` reference pair runs 2 chunks

## Documentation

- [ ] `content/tutorials/scenarios_and_models.md` (asked for in #80)

## Relationships

- Parent (theme epic): #204
- Blocked by: `build_basin.py --source real` (under #204) for real data; runnable earlier on the synthetic bundle
- Blocks: #78, #79, #80
- Related: Cartesian-adapter issue (under #204)
